org-code-javabuilder/lib/src/main/java/dev/javabuilder/WebSocketServer.java

```diff
@@ -50,22 +53,27 @@ public void onOpen(Session session) {
     final String connectionId = "LocalhostWebSocketConnection";
     final String levelId = queryInput.getString("level_id");
     final String channelId = queryInput.getString("channel_id");
+    final ExecutionType executionType =
+        ExecutionType.valueOf(queryInput.getString("execution_type"));
     final String dashboardHostname = "http://" + queryInput.get("iss") + ":3000";
     final JSONObject options = new JSONObject(queryInput.getString("options"));
     boolean useNeighborhood = false;
     if (options.has("useNeighborhood")) {
       String useNeighborhoodStr = options.getString("useNeighborhood");
       useNeighborhood = Boolean.parseBoolean(useNeighborhoodStr);
     }
-
-    final String executionTypeString =
-        queryInput.has("execution_type") ? queryInput.getString("execution_type") : null;
-    // TODO: in order to not break current behavior in Javalab, execution type defaults to 'RUN'.
-    // Once Javalab is sending the execution type parameter, remove this fallback
-    final ExecutionType executionType =
-        executionTypeString == null
-            ? ExecutionType.RUN
-            : ExecutionType.valueOf(executionTypeString);
+    final List<String> compileList;
+    if (options.has("compileList")) {
+      compileList =
+          options
+              .getJSONArray("compileList")
+              .toList()
+              .stream()
+              .map(filename -> (String) filename)
+              .collect(Collectors.toList());
+    } else {
+      compileList = new ArrayList<>();
```

Comment:
it feels weird to send an empty list for running code when we actually want to compile everything. Would it make more sense to send `null` instead?

Reply:
I'd lean towards passing null and handling it on the method side, because it's a little clearer.